In [1]:
import pandas as pd

dataset = pd.read_csv('breast-cancer.csv')

dataset.shape

(285, 10)

In [2]:
dataset

'40-49'  'premeno'  '15-19'  '0-2'  'yes'  '3'  'right'   'left_up'  \
0    '50-59'     'ge40'  '15-19'  '0-2'   'no'  '1'  'right'   'central'   
1    '50-59'     'ge40'  '35-39'  '0-2'   'no'  '2'   'left'  'left_low'   
2    '40-49'  'premeno'  '35-39'  '0-2'  'yes'  '3'  'right'  'left_low'   
3    '40-49'  'premeno'  '30-34'  '3-5'  'yes'  '2'   'left'  'right_up'   
4    '50-59'  'premeno'  '25-29'  '3-5'   'no'  '2'  'right'   'left_up'   
..       ...        ...      ...    ...    ...  ...      ...         ...   
280  '50-59'     'ge40'  '30-34'  '6-8'  'yes'  '2'   'left'  'left_low'   
281  '50-59'  'premeno'  '25-29'  '3-5'  'yes'  '2'   'left'  'left_low'   
282  '30-39'  'premeno'  '30-34'  '6-8'  'yes'  '2'  'right'  'right_up'   
283  '50-59'  'premeno'  '15-19'  '0-2'   'no'  '2'  'right'  'left_low'   
284  '50-59'     'ge40'  '40-44'  '0-2'   'no'  '3'   'left'  'right_up'   

      'no'     'recurrence-events'  
0     'no'  'no-recurrence-events'  
1     'no'     'recurrence-events'  
2    'yes'  'no-recurrence-events'  
3     'no'     'recurrence-events'  
4    'yes'  'no-recurrence-events'  
..     ...                     ...  
280   'no'  'no-recurrence-events'  
281  'yes'  'no-recurrence-events'  
282   'no'  'no-recurrence-events'  
283   'no'  'no-recurrence-events'  
284   'no'  'no-recurrence-events'  

[285 rows x 10 columns]

In [3]:
#merapikan dan memberikan keterangan pada dataset
dataset = pd.read_csv('breast-cancer.csv', sep = ',', names=['age','menopause','tumor-size','inv-nodes','node-caps','deg-malig','breast','breast-quad','irradiat','Class'])

dataset.shape

(286, 10)

In [4]:
dataset

age  menopause tumor-size inv-nodes node-caps deg-malig   breast  \
0    '40-49'  'premeno'    '15-19'     '0-2'     'yes'       '3'  'right'   
1    '50-59'     'ge40'    '15-19'     '0-2'      'no'       '1'  'right'   
2    '50-59'     'ge40'    '35-39'     '0-2'      'no'       '2'   'left'   
3    '40-49'  'premeno'    '35-39'     '0-2'     'yes'       '3'  'right'   
4    '40-49'  'premeno'    '30-34'     '3-5'     'yes'       '2'   'left'   
..       ...        ...        ...       ...       ...       ...      ...   
281  '50-59'     'ge40'    '30-34'     '6-8'     'yes'       '2'   'left'   
282  '50-59'  'premeno'    '25-29'     '3-5'     'yes'       '2'   'left'   
283  '30-39'  'premeno'    '30-34'     '6-8'     'yes'       '2'  'right'   
284  '50-59'  'premeno'    '15-19'     '0-2'      'no'       '2'  'right'   
285  '50-59'     'ge40'    '40-44'     '0-2'      'no'       '3'   'left'   

    breast-quad irradiat                   Class  
0     'left_up'     'no'     'recurrence-events'  
1     'central'     'no'  'no-recurrence-events'  
2    'left_low'     'no'     'recurrence-events'  
3    'left_low'    'yes'  'no-recurrence-events'  
4    'right_up'     'no'     'recurrence-events'  
..          ...      ...                     ...  
281  'left_low'     'no'  'no-recurrence-events'  
282  'left_low'    'yes'  'no-recurrence-events'  
283  'right_up'     'no'  'no-recurrence-events'  
284  'left_low'     'no'  'no-recurrence-events'  
285  'right_up'     'no'  'no-recurrence-events'  

[286 rows x 10 columns]

In [5]:
dataset.info()

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 286 entries, 0 to 285
Data columns (total 10 columns):
 #   Column       Non-Null Count  Dtype 
---  ------       --------------  ----- 
 0   age          286 non-null    object
 1   menopause    286 non-null    object
 2   tumor-size   286 non-null    object
 3   inv-nodes    286 non-null    object
 4   node-caps    278 non-null    object
 5   deg-malig    286 non-null    object
 6   breast       286 non-null    object
 7   breast-quad  285 non-null    object
 8   irradiat     286 non-null    object
 9   Class        286 non-null    object
dtypes: object(10)
memory usage: 22.5+ KB


In [6]:
dataset = dataset.drop(columns=['age','menopause','inv-nodes','deg-malig','breast','breast-quad','irradiat','Class'])

dataset

tumor-size node-caps
0      '15-19'     'yes'
1      '15-19'      'no'
2      '35-39'      'no'
3      '35-39'     'yes'
4      '30-34'     'yes'
..         ...       ...
281    '30-34'     'yes'
282    '25-29'     'yes'
283    '30-34'     'yes'
284    '15-19'      'no'
285    '40-44'      'no'

[286 rows x 2 columns]

In [7]:
dataset.info()

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 286 entries, 0 to 285
Data columns (total 2 columns):
 #   Column      Non-Null Count  Dtype 
---  ------      --------------  ----- 
 0   tumor-size  286 non-null    object
 1   node-caps   278 non-null    object
dtypes: object(2)
memory usage: 4.6+ KB


In [8]:
dataset = dataset.dropna()

In [9]:
dataset.info()

<class 'pandas.core.frame.DataFrame'>
Int64Index: 278 entries, 0 to 285
Data columns (total 2 columns):
 #   Column      Non-Null Count  Dtype 
---  ------      --------------  ----- 
 0   tumor-size  278 non-null    object
 1   node-caps   278 non-null    object
dtypes: object(2)
memory usage: 6.5+ KB


In [10]:
#membuat contigency table
contigency_table = pd.crosstab(index=dataset['tumor-size'], columns=dataset['node-caps'], margins=True)

contigency_table

node-caps   'no'  'yes'  All
tumor-size                  
'0-4'          8      0    8
'10-14'       27      1   28
'15-19'       25      4   29
'20-24'       39      9   48
'25-29'       41     10   51
'30-34'       42     16   58
'35-39'       12      7   19
'40-44'       16      6   22
'45-49'        2      1    3
'5-9'          4      0    4
'50-54'        6      2    8
All          222     56  278

In [11]:
#menghitung setiap expected value
#'0-4' yes
p = (56/278)*(8/278)
ev1 = 278*p

In [12]:
#menghitung setiap expected value
#'0-4' no
p = (222/278)*(8/278)
ev2 = 278*p
ev2

6.388489208633094

In [13]:
#'5-9' yes
p = (56/278)*(4/278)
ev3 = 278*p
ev3

0.8057553956834532

In [14]:
#'5-9' no
p = (222/278)*(4/278)
ev4 = 278*p
ev4

3.194244604316547

In [15]:
#'10-14' yes
p = (56/278)*(28/278)
ev5 = 278*p
ev5

5.640287769784173

In [16]:
#'10-14' no
p = (222/278)*(28/278)
ev6 = 278*p
ev6

22.35971223021583

In [17]:
#'15-19' yes
p = (56/278)*(29/278)
ev7 = 278*p
ev7

5.841726618705036

In [18]:
#'15-19' no
p = (222/278)*(29/278)
ev8 = 278*p
ev8

23.158273381294965

In [19]:
#'20-24' yes
p = (56/278)*(48/278)
ev9 = 278*p
ev9

9.66906474820144

In [20]:
#'20-24' no
p = (222/278)*(48/278)
ev10 = 278*p
ev10

38.330935251798564

In [21]:
#'25-29' yes
p = (56/278)*(51/278)
ev11 = 278*p
ev11

10.273381294964029

In [22]:
#'25-29' no
p = (222/278)*(51/278)
ev12 = 278*p
ev12

40.726618705035975

In [23]:
#'30-34' yes
p = (56/278)*(58/278)
ev13 = 278*p
ev13

11.683453237410072

In [24]:
#'30-34' no
p = (222/278)*(58/278)
ev14 = 278*p
ev14

46.31654676258993

In [25]:
#'35-39' yes
p = (56/278)*(19/278)
ev15 = 278*p
ev15

3.8273381294964035

In [26]:
#'35-39' no
p = (222/278)*(19/278)
ev16 = 278*p
ev16

15.1726618705036

In [27]:
#'40-44' yes
p = (56/278)*(22/278)
ev17 = 278*p
ev17

4.431654676258992

In [28]:
#'40-44' no
p = (222/278)*(22/278)
ev18 = 278*p
ev18

17.568345323741006

In [29]:
#'45-49' yes
p = (56/278)*(3/278)
ev19 = 278*p
ev19

0.60431654676259

In [30]:
#'45-49' no
p = (222/278)*(3/278)
ev20 = 278*p
ev20

2.3956834532374103

In [31]:
#'50-54' yes
p = (56/278)*(8/278)
ev21 = 278*p
ev21

1.6115107913669064

In [32]:
#'50-54' no
p = (222/278)*(8/278)
ev22 = 278*p
ev22

6.388489208633094

In [33]:
#menghitung nilai statistik chi-square
chi_square = []

In [34]:
#'0-4' yes
OE1 = 0-ev1
square_of_oe1 = OE1**2
chi_square.append(square_of_oe1/ev1)

In [35]:
#'0-4' no
OE2 = 8-ev2
square_of_oe2 = OE2**2
chi_square.append(square_of_oe2/ev2)

In [36]:
#'5-9' yes
OE3 = 0-ev3
square_of_oe3 = OE3**2
chi_square.append(square_of_oe3/ev3)

In [37]:
#'5-9' no
OE4 = 4-ev4
square_of_oe4 = OE4**2
chi_square.append(square_of_oe4/ev4)

In [38]:
#'10-14' yes
OE5 = 1-ev5
square_of_oe5 = OE5**2
chi_square.append(square_of_oe5/ev5)

In [39]:
#'10-14' no
OE6 = 27-ev6
square_of_oe6 = OE6**2
chi_square.append(square_of_oe6/ev6)

In [40]:
#'15-19' yes
OE7 = 4-ev7
square_of_oe7 = OE7**2
chi_square.append(square_of_oe7/ev7)

In [41]:
#'15-19' no
OE8 = 25-ev8
square_of_oe8 = OE8**2
chi_square.append(square_of_oe8/ev8)

In [42]:
#'20-24' yes
OE9 = 9-ev9
square_of_oe9 = OE9**2
chi_square.append(square_of_oe9/ev9)

In [43]:
#'20-24' no
OE10 = 39-ev10
square_of_oe10 = OE10**2
chi_square.append(square_of_oe10/ev10)

In [44]:
#'25-29' yes
OE11 = 10-ev11
square_of_oe11 = OE11**2
chi_square.append(square_of_oe11/ev11)

In [45]:
#'25-29' no
OE12 = 41-ev12
square_of_oe12 = OE12**2
chi_square.append(square_of_oe12/ev12)

In [46]:
#'30-34' yes
OE13 = 16-ev13
square_of_oe13 = OE13**2
chi_square.append(square_of_oe13/ev13)

In [47]:
#'30-34' no
OE14 = 42-ev14
square_of_oe14 = OE14**2
chi_square.append(square_of_oe14/ev14)

In [48]:
#'35-39' yes7
OE15 = 7-ev15
square_of_oe15 = OE15**2
chi_square.append(square_of_oe15/ev15)

In [49]:
#'35-39' no
OE16 = 12-ev16
square_of_oe16 = OE16**2
chi_square.append(square_of_oe16/ev16)

In [50]:
#'40-44' yes
OE17 = 6-ev17
square_of_oe17 = OE17**2
chi_square.append(square_of_oe17/ev17)

In [51]:
#'40-44' no
OE18 = 16-ev18
square_of_oe18 = OE18**2
chi_square.append(square_of_oe18/ev18)

In [52]:
#'45-49' yes
OE19 = 1-ev19
square_of_oe19 = OE19**2
chi_square.append(square_of_oe19/ev19)

In [53]:
#'45-49' no
OE20 = 2-ev20
square_of_oe20 = OE20**2
chi_square.append(square_of_oe20/ev20)

In [54]:
#'50-54' yes
OE21 = 2-ev21
square_of_oe21 = OE21**2
chi_square.append(square_of_oe21/ev21)

In [55]:
#'50-54' no
OE22 = 6-ev22
square_of_oe22 = OE22**2
chi_square.append(square_of_oe22/ev22)

In [56]:
chi_square

[1.6115107913669064,
 0.4065072266511114,
 0.8057553956834532,
 0.2032536133255557,
 3.81758368815152,
 0.9629940834976792,
 0.5806428748626715,
 0.1464684729383314,
 0.04629689105858174,
 0.01167849504180428,
 0.007274852386997969,
 0.0018350978994228632,
 1.5947832866711555,
 0.4022876759170485,
 2.62996970844377,
 0.6634157823101415,
 0.555031299635617,
 0.14000789540357872,
 0.25907845152449466,
 0.0653531229070798,
 0.09365364850976364,
 0.023624343768228688]

In [57]:
chi_square_value = sum(chi_square)

chi_square_value

15.029006697954912

Pada hasil perhitungan didapat bahwa chi square value = 15.029006697954912

degree of freedom = (11-1)*(2-1) = 10

Jika selang tingkat kepercayaan yang kita gunakan adalah 95%, maka nilai 𝛼 = 0.05
berdasarkan tabel persentase nilai distribusi chi square (dengan degree of freedom = 10 dan 𝛼 = 0.05)nilai/batasnya = 18.31

Maka, nilai chi square bisa diterima karena <18.31 atau dengan kata lain berada pada accepted region.